File: shared/src/main/java/xyz/klinker/messenger/shared/activity/AbstractSettingsActivity.java
package xyz.klinker.messenger.shared.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import xyz.klinker.messenger.shared.R;

public class AbstractSettingsActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_settings);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
    }

    public Toolbar getToolbar() {
        return (Toolbar) findViewById(R.id.toolbar);
    }
}

File: shared/src/main/java/xyz/klinker/messenger/shared/activity/AbstractSettingsActivity.kt
package xyz.klinker.messenger.shared.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.View

import xyz.klinker.messenger.shared.R

open class AbstractSettingsActivity : AppCompatActivity() {

    val toolbar: Toolbar? by lazy { findViewById<View>(R.id.toolbar) as Toolbar? }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_settings)
        if (toolbar != null) {
            setSupportActionBar(toolbar)
        }
    }
}
